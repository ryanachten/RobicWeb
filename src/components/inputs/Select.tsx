import React from "react";
import FormControl from "@material-ui/core/FormControl";
import InputLabel from "@material-ui/core/InputLabel";
import MenuItem from "@material-ui/core/MenuItem";
import MuiSelect from "@material-ui/core/Select";

type Option = {
  id: string;
  value: any;
  label: string;
};

type Props = {
  label: string;
  className?: string;
  options: Option[];
  onChange: (e: any) => void;
  value: any;
};

export const Select = ({
  className,
  label,
  options,
  onChange,
  value
}: Props) => (
  <FormControl className={className}>
    <InputLabel htmlFor={label}>{label}</InputLabel>
    <MuiSelect
      onChange={onChange}
      value={value}
      inputProps={{
        name: label,
        id: label
      }}
    >
      {options &&
        options.map((option: Option) => (
          <MenuItem key={option.id} value={option.value}>
            {option.label}
          </MenuItem>
        ))}
    </MuiSelect>
  </FormControl>
);
