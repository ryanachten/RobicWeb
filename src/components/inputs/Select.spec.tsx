import React from "react";
import { createShallow } from "@material-ui/core/test-utils";
import { Select } from "./Select";
import renderer from "react-test-renderer";

describe("Component: Select", () => {
  let shallow: any;

  beforeEach(() => {
    shallow = createShallow({ untilSelector: "Select" });
  });

  const initProps = {
    label: "Testing",
    options: [
      {
        id: "test",
        value: "test",
        label: "test"
      }
    ],
    onChange: jest.fn(),
    value: "test"
  };

  it("renders correctly", () => {
    const tree = renderer.create(<Select {...initProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
