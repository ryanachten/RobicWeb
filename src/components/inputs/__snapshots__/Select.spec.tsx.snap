// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Select renders correctly 1`] = `
<div
  className="MuiFormControl-root-1"
>
  <label
    className="MuiFormLabel-root-16 MuiFormLabel-filled-20 MuiInputLabel-root-5 MuiInputLabel-formControl-10 MuiInputLabel-animated-13 MuiInputLabel-shrink-12"
    data-shrink={true}
    htmlFor="Testing"
  >
    Testing
  </label>
  <div
    className="MuiInputBase-root-43 MuiInput-root-30 MuiInput-underline-34 MuiInputBase-formControl-44 MuiInput-formControl-31"
    onClick={[Function]}
  >
    <div
      className="MuiSelect-root-23"
    >
      <div
        aria-haspopup="true"
        aria-pressed="false"
        className="MuiSelect-select-24 MuiSelect-selectMenu-27 MuiInputBase-input-53 MuiInput-input-38"
        id="select-Testing"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        test
      </div>
      <input
        id="Testing"
        name="Testing"
        type="hidden"
        value="test"
      />
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-60 MuiSelect-icon-29"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
</div>
`;
